src/gui/plugins/apply_force_torque/ApplyForceTorque.cc

```diff
@@ -0,0 +1,336 @@
+        static_cast<gz::sim::gui::events::EntitiesSelected*>(_event);
+    this->dataPtr->selectedEntities = _e->Data();
+    this->dataPtr->changedEntity = true;
+  }
+  else if (_event->type() ==
+    gz::sim::gui::events::DeselectAllEntities::kType)
+  {
+    this->dataPtr->selectedEntities.clear();
+    this->dataPtr->changedEntity = true;
+  }
+
+  return QObject::eventFilter(_obj, _event);
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::Update(const UpdateInfo &/*_info*/,
+    EntityComponentManager &_ecm)
+{
+  {
+    std::lock_guard<std::mutex> lock(this->dataPtr->mutex);
+
+    if (!this->dataPtr->changedEntity)
+    {
+      return;
+    }
+    this->dataPtr->changedEntity = false;
+
+    this->dataPtr->modelName = "";
+    this->dataPtr->linkList.clear();
+    this->dataPtr->linkNameList.clear();
+    this->dataPtr->linkIndex = -1;
+
+    if (!this->dataPtr->selectedEntities.empty())
+    {
+      auto entity = this->dataPtr->selectedEntities.front();
+      Model selectedModel(entity);
+      Link selectedLink(entity);
+      if (selectedModel.Valid(_ecm))
+      {
+        selectedLink = Link(selectedModel.CanonicalLink(_ecm));
+      }
+      else if (selectedLink.Valid(_ecm))
+      {
+        selectedModel = *selectedLink.ParentModel(_ecm);
+      }
+      else
+      {
+        return;
+      }
+
+      this->dataPtr->modelName = QString::fromStdString(
+        selectedModel.Name(_ecm));
+
+      // Put all of the model's links into the list
+      auto links = selectedModel.Links(_ecm);
+      unsigned int i{0};
+      while (i < selectedModel.LinkCount(_ecm))
+      {
+        Link link(links[i]);
+        this->dataPtr->linkList.push_back(link);
+        this->dataPtr->linkNameList.push_back(
+          QString::fromStdString(*link.Name(_ecm)));
+        if (link.Entity() == selectedLink.Entity())
+        {
+          this->dataPtr->linkIndex = i;
+        }
+        ++i;
+      }
+
+      // Create publisher if not yet created
+      if (!this->dataPtr->pub.Valid())
+      {
+        std::string worldName{""};
+        _ecm.Each<components::World, components::Name>(
+            [&](const Entity &_entity,
+              const components::World * /* _world */ ,
+              const components::Name *_name)->bool
+            {
+              World world(_entity);
+              for (auto &model : world.Models(_ecm))
+              {
+                if (model == entity)
+                {
+                  worldName = _name->Data();
+                  return true;
+                }
+              }
+              gzerr << "World not found" << std::endl;
+              return false;
+            });
+        auto topic = transport::TopicUtils::AsValidTopic(
+          "/world/" + worldName + "/wrench");
+        this->dataPtr->pub =
+            this->dataPtr->node.Advertise<msgs::EntityWrench>(topic);
+        gzdbg << "Created publisher to " << topic << std::endl;
```

Comment:
Creation of the publisher should be done in `LoadConfig` and you can get the worldName using `gui::worldNames` similar to https://github.com/gazebosim/gz-sim/blob/064f69ed7c441b90abbf0f84eeb0bfaeec628da9/src/gui/plugins/spawn/Spawn.cc#L186-L188